src/components/album/AlbumChatBox.js

```diff
@@ -47,28 +47,44 @@ const AlbumChatBox = ({ spotifyUrl }) => {
   };
 
   useEffect(() => {
+    const generateMessageWithKey = (data) => {
+      const isOwnMessage = loggedInUserId === data.user._id;
+      return {
+        ...data,
+        isOwnMessage,
+        messageKey: uuidv4(),
+      };
+    };
+
+    const fetchRecentMessages = async () => {
+      const response = await fetch(
+        `${process.env.REACT_APP_API_BASE_PATH}/chat/${albumId}`
+      );
+      const data = await response.json();
+      const messagesWithKeys = data.messages
+        .map((message) => ({
+          ...message,
+          user: { name: message.user },
+        }))
+        .map(generateMessageWithKey);
+      setMessages(messagesWithKeys);
+    };
+
     if (albumId && !socket) {
       const newSocket = io(process.env.REACT_APP_SOCKET_BASE_PATH);
       newSocket.on('connect', () => {
         console.log('Connected to WebSocket server');
-        // Request to join the album chat room
         newSocket.emit('join:album_chat', albumId);
         newSocket.on('chat:album', (data) => {
-          console.log('Received message from chat:album:', data);
-
-          const isOwnMessage = loggedInUserId === data.user._id; // Determine isOwnMessage
-          const messageWithKey = {
-            ...data,
-            isOwnMessage,
-            messageKey: uuidv4(),
-          }; // will add key to the message
+          const messageWithKey = generateMessageWithKey(data);
           setMessages((prevMessages) => [...prevMessages, messageWithKey]);
-
           scrollToBottom();
         });
       });
       setSocket(newSocket);
     }
+
+    fetchRecentMessages();
```

Comment:
`useEffect` runs everytime any of its dependencies change: https://react.dev/reference/react/useEffect#specifying-reactive-dependencies

For this `useEffect` we have told React that it depends on: `albumId`, `socket`, and `loggedInUserId`.

When the component first loads, it will get the `albumId` by doing regex on the `spotifyUrl` prop. It will also get the logged in user from the `useAuth` hook. But `socket` will be undefined at this point.

This`useEffect` will run for the first time when the component is mounted. Then, because `socket` was undefined we create it and set up all our listeners and whatnot, and then set the socket on line `84`. This means the `socket` variable has changed from `undefined` -> a populated value. This triggers this `useEffect` to run again.

This time, the code inside `albumId && !socket` won't run again since `socket` now has a value; but it **will** run `fetchRecentMessages` again.

We can see this in our dev tools that the messages are being fetched 2 times from the chat endpoint:
<img width="838" alt="Screenshot 2023-11-06 at 2 43 15 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/d023e48e-fff3-41db-ba20-667ad95602de">

---

It's not a huge deal right now but as our site grows, that means every user who enters the chat room, their browser is going to make 2 calls to the backend server, when 1 would have been enough. This wastes our server resources and means we are using more of the user's bandwidth than we need to.

Since we only want to fetch messages, when the user first loads the components, and after that we can rely on the live messages from the socket, we can move the `fetchMessages` code into a separate `useEffect` that has an empty dependencies array so that it only runs when the component is first loaded. 

Alternatively, we can add an `if` statement here so that `fetchMessages` only runs when the socket is not yet setup. The downside with that is we're starting to have a large `useEffect` that is doing many different things. So it's a good sign to start breaking it up in to smaller `useEffect`s